File: LeetCode/Tag/Sort/cpp/56.merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort (intervals.begin(), intervals.end(), [] (vector<int>& i1, vector<int>& i2) {
            return i1[0] == i2[0] ? i1[1] > i2[1] : i1[0] < i2[0];
        });

        vector<vector<int>> res (1, intervals[0]);
        for (int i = 1; i < intervals.size(); ++i) {
            if (res.back()[1] < intervals[i][1]) {
                if (res.back()[1] < intervals[i][0]) {
                    res.push_back(intervals[i]);
                } else {
                    res.back()[1] = intervals[i][1];
                }
            }
        }
        return res;
    }
};

File: LeetCode/Tag/Array/js/56.merge-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    intervals = intervals.sort((a,b)=> a[0] - b[0]);
    for(let i=1;i<intervals.length;i++){
        const [prevStart, prevEnd] = intervals[i-1];
        const [currStart, currEnd] = intervals[i];

        if(prevEnd >= currStart){
            // remove prev value
            // update current value with new intervals

            intervals[i-1] = null;
            intervals[i][0] = Math.min(prevStart,currStart);
            intervals[i][1] = Math.max(prevEnd,currEnd);
        }
    }
    return intervals.filter(i=>!!i);
};
